clients/drcachesim/tests/schedule_stats_test.cpp

```diff
@@ -324,12 +341,92 @@ test_cpu_footprint()
     return true;
 }
 
+static bool
+test_syscall_latencies()
+{
+    static constexpr int64_t TID_A = 42;
+    static constexpr int64_t TID_B = 142;
+    static constexpr int64_t TID_C = 242;
+    static constexpr uintptr_t SYSNUM_X = 12;
+    static constexpr uintptr_t SYSNUM_Y = 167;
+    std::vector<std::vector<memref_t>> memrefs = {
+        {
+            gen_instr(TID_A),
+            // Involuntary switch.
+            gen_instr(TID_B),
+            gen_marker(TID_B, TRACE_MARKER_TYPE_TIMESTAMP, 1100),
+            gen_marker(TID_B, TRACE_MARKER_TYPE_SYSCALL, SYSNUM_X),
+            gen_marker(TID_B, TRACE_MARKER_TYPE_TIMESTAMP, 1600),
+            // Voluntary switch, on non-maybe-blocking-marked syscall.
```

Comment:
Append: Latency 500.

And elsewhere too.

Reply:
Done.